### PR TITLE
Issue #253 - Newtab not Modified

### PR DESCRIPTION
**Pull request by FrancescoSTL**
@schalkneethling r?

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1117,4 +1136,9 @@ exports.main = function() {
     if(typeof simpleStorage.seenUserData === 'undefined') {
         modifyNewtab();
     }
+
+    var resetPreloadTimer = timers.setTimeout(function() {
```
Comment (FrancescoSTL) [inline index.js]: Probably should've noted here that this is the failsafe which should reset the preference after 24 hours if it isn't already reset.

File: index.js [hunk 2]
```diff
@@ -959,7 +963,6 @@ function modifyNewtab() {
             // if we couldn't check if we can do the auto import because we weren't able to run the canUndo function, throw an error, and don't modify the newtab page with anything
             } catch(e) {
                 console.error('Not able to resolve autoimport undo promise.' + e);
-                worker.port.emit('modify', headerContent, footerContent);
```
Comment (FrancescoSTL) [inline index.js]: Removing, as this was there for debugging and I forgot to remove it in the last PR.

File: index.js [hunk 3]
```diff
@@ -927,6 +930,7 @@ function modifyNewtab() {
     aboutNewtab = pageMod.PageMod({
         include: /about:newtab/,
         contentScriptFile: './js/about-newtab.js',
+        contentScriptWhen: 'ready',
```
Comment (FrancescoSTL) [inline index.js]: Re-adding this line, because it doesn't seem to fix the problem in v4.

File: index.js [hunk 4]
```diff
@@ -1050,6 +1059,16 @@ function destroy() {
     }
 }
 
+function resetPreload() {
+    // Check to see if the newtab is not being preloaded
+    // note: The binary that this addon will be packaged with will have browser.newtab.preload set to false. 
+    //       This is to mitigate the cache overriding our pagemod when users load newtab.
```
Comment (schalkneethling) [inline index.js]: Nit: indent

**Comment by schalkneethling** [comment]
r+

**Pull request merged**